chore: Bump CAPI to v1.4.3

Ankitasw opened the pull request:
<!-- Thanks for this PR! If this is your first PR please read the [contributing guide](../CONTRIBUTING.md) -->
<!-- If this PR is still work-in-progress and is being open for visibility please prefix the title with `WIP:` -->

**What type of PR is this?**
/kind support


**What this PR does / why we need it**:
This PR bumps CAPI to v1.4.3.

<!-- Enter a description of the change and why this change is needed -->

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:
<!-- Put an "X" character inside the brackets of each completed task. Some may be optional depending on the PR in which case these can be deleted -->

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
Bump CAPI to v1.4.3
```

Ankitasw commented:
/retest

Ankitasw commented:
/test pull-cluster-api-provider-aws-e2e-blocking

Ankitasw commented:
/test pull-cluster-api-provider-aws-e2e-blocking

Ankitasw commented:
/test pull-cluster-api-provider-aws-e2e-blocking

Ankitasw commented:
/test pull-cluster-api-provider-aws-e2e-eks

Ankitasw commented:
/test pull-cluster-api-provider-aws-e2e

Ankitasw commented:
/cherry-pick release-2.1

Ankitasw commented:
PR is up for review.

Ankitasw commented:
/retest

richardcase commented on bootstrap/eks/controllers/eksconfig_controller.go:
```diff
@@ -212,26 +212,26 @@ func (r *EKSConfigReconciler) joinWorker(ctx context.Context, cluster *clusterv1
 		log.Info("Control Plane has not yet been initialized")
 		conditions.MarkFalse(config, eksbootstrapv1.DataSecretAvailableCondition, eksbootstrapv1.WaitingForControlPlaneInitializationReason, clusterv1.ConditionSeverityInfo, "")
-		return ctrl.Result{}, nil
+		return nil
```
Would we not want to requeue in this scenario? (i know we didn't originally).

Ankitasw replied:
we can move this discussion to separate PR

richardcase commented on exp/controllers/awsmachinepool_controller.go:
```diff
@@ -200,28 +200,28 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 		machinePoolScope.Info("Cluster infrastructure is not ready yet")
 		conditions.MarkFalse(machinePoolScope.AWSMachinePool, expinfrav1.ASGReadyCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
-		return ctrl.Result{}, nil
+		return nil
```
We should be requeuing in this scenario.

richardcase commented on exp/controllers/awsmachinepool_controller.go:
```diff
@@ -200,28 +200,28 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 		machinePoolScope.Info("Bootstrap data secret reference is not yet available")
 		conditions.MarkFalse(machinePoolScope.AWSMachinePool, expinfrav1.ASGReadyCondition, infrav1.WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
-		return ctrl.Result{}, nil
+		return nil
```
And we should requeue/requeueafter here

richardcase reviewed (commented):
@Ankitasw - there are a lot of changes in here that aren't specifically related to bumping CAPI to v1.4.3.

I'd say that we shouldn't change the return values of the reconcoleNormal/reconcileDelete functions to remove the ctrl.Result for 2 reasons:
- its the pattern used widely across CAPI
- There are some scenarios where we should be requeueing (and we're currently not).

Ankitasw commented:
@richardcase I think they dealt in CAPI the same way, isn't it?
You can see changes here: https://github.com/kubernetes-sigs/cluster-api/pull/8796/files#diff-47cf5969a7a97b70232c5d97b882424745bd1ed511b29aa56f55194688ab833c

I understand that not all changes are related to CAPI bump, shall I revert golangci bump changes and raise another PR?

Ankitasw commented:
/test pull-cluster-api-provider-aws-e2e
/test pull-cluster-api-provider-aws-e2e-eks

Ankitasw commented:
/test pull-cluster-api-provider-aws-e2e
/test pull-cluster-api-provider-aws-e2e-eks

Ankitasw commented:
PR is ready for review.

richardcase commented:
Just seen that you reverted some of the changes, sorry @Ankitasw i should've pinged you on slack to discuss.

As it stands now and with the e2e passing:

/approve
/lgtm

Ankitasw commented:
> Looking through the logic in the reconcileDelete/Normal, I think there are situations where we should be requeuing, and we aren't currently, so if we removed the result, then we would have to add them back in if we wanted to change these to requeue. But agree as it stands with no requeues then having the extra return value is pointless.

Addressed this in #4326
